src/compiler/factory.ts

```diff
@@ -2798,6 +2798,20 @@ namespace ts {
         return tokenSourceMapRanges && tokenSourceMapRanges[token];
     }
 
+    export function getConstantValue(node: Node) {
+        const emitNode = node.emitNode;
+        if (emitNode && emitNode.flags & EmitFlags.ConstantValue) {

```

Comment:
q: why can't you just check if there exists emitNode.constantValue instead of using the EmitFlags.ConstatntValue